Apps/Examples/Examples/All Examples/AdvancedViewportGesturesExample.swift

```diff
@@ -0,0 +1,139 @@
+import UIKit
+@_spi(Experimental) import MapboxMaps
+
+@objc(AdvancedViewportGesturesExample)
+final class AdvancedViewportGesturesExample: UIViewController, ExampleProtocol {
+
+    private enum State {
+        case following
+        case overview
+    }
+
+    private var state: State = .following {
+        didSet {
+            syncWithState()
+        }
+    }
+
+    private var mapView: MapView!
+    private var followPuckViewportState: FollowPuckViewportState!
+    private var overviewViewportState: OverviewViewportState!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        mapView = MapView(frame: view.bounds)
+        view.addSubview(mapView)
+
+        let cupertino = CLLocationCoordinate2D(latitude: 37.3230, longitude: -122.0322)
+
+        mapView.mapboxMap.setCamera(to: CameraOptions(center: cupertino, zoom: 14))
+
+        mapView.location.options.puckType = .puck2D(.makeDefault(showBearing: true))
+        mapView.location.options.puckBearingSource = .course
+
+        followPuckViewportState = mapView.viewport.makeFollowPuckViewportState(
+            options: FollowPuckViewportStateOptions(
+                bearing: .course))
+
+        overviewViewportState = mapView.viewport.makeOverviewViewportState(
```

Comment:
seems you are using the real device's location for this example, while on Android, we have a custom LocationProvider implementation to emits fake location updates on a route. 
I think it is ok to be different on iOS, as the main showcase is how to customise gestures to work with the viewport, but would be good to consider adding the route simulator later on.

Reply:
Added some comments to these examples explaining how to simulate location in the iOS simulator